source/manuals/programming-languages/nodejs/index.html.md.erb

```diff
@@ -395,11 +395,24 @@ that a Node developer would have trouble reading your code is advised against.
 
 ## Frameworks
 
-> Use Express for web applications, avoid lesser-known frameworks
+It is important to remember that we use the
+[govuk-frontend](https://github.com/alphagov/govuk-frontend) to build pages on websites
+at GDS. This restricts the choice of webservers to ones that easily support rendering
+content using [Nunjucks](https://mozilla.github.io/nunjucks/). This rules out many
+currently popular webservers that use React or other templating approaches.
+
+> Use Koa or Express for web applications, avoid lesser-known frameworks
```

Comment:
I was collecting evidence for this content update, and I wonder if this choice is a little bit out of date?

Without wanting to propose a proliferation of frameworks, I wonder if [Hapi](https://hapi.dev/tutorials/expresstohapi/) is a closer fit for GDS ways of working as it comes with many of the expected features from Express (including routing) and works in a similar way to Express middleware.

Koa, while created by the initial Express author, uses JavaScript generators which are not in common usage amongst either browser or Node.js developers.